// pragmate-ui/preload/preload.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/styles';
import * as dependency_2 from 'react';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"pragmate-ui@0.0.6/preload"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1],['react', dependency_2]]);

brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/preload')


const ims = new Map();

/***********************
INTERNAL MODULE: ./index
***********************/

ims.set('./index', {hash: 2002719117, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Preload = Preload;
var _react = require("react");
/*bundle */function Preload(props) {
  const {
    width,
    height,
    color,
    stopAnimation,
    radio
  } = props;
  return _react.default.createElement("span", {
    className: 'pui-preload',
    style: {
      width: width || '100%',
      height: height || '14px',
      background: color || '',
      borderRadius: radio || ''
    }
  }, _react.default.createElement("span", {
    style: {
      display: stopAnimation ? 'none' : 'block'
    }
  }));
}
}});

__pkg.exports.descriptor = [{"im":"./index","from":"Preload","name":"Preload"}];

export let Preload;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Preload') && (Preload = require ? require('./index').Preload : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
