// pragmate-ui/dropdown-bottom-menu/dropdown-bottom-menu.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react", "pragmate-ui/image", "pragmate-ui/icons"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3, dependency_4) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.DropdownBottomMenu = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "pragmate-ui@0.0.6/dropdown-bottom-menu"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2], ['pragmate-ui/image', dependency_3], ['pragmate-ui/icons', dependency_4]]);
  brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/dropdown-bottom-menu');
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./index
  ***********************/

  ims.set('./index', {
    hash: 2794062376,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.DropdownBottomMenu = DropdownBottomMenu;
      var _react = require("react");
      var _image = require("pragmate-ui/image");
      var _list = require("./list");
      /*bundle*/
      function DropdownBottomMenu({
        handleClick,
        show,
        items,
        children,
        isImage,
        title
      }) {
        const cls = show ? "pull-down-menu" : "pull-down-menu pull-down-menu__hide";
        const isItems = !!items && Array.isArray(items);
        return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("div", {
          className: cls
        }, (title || isImage) && _react.default.createElement("header", null, isImage && _react.default.createElement("div", {
          className: "image-menu"
        }, _react.default.createElement(_image.Image, {
          src: `${globalThis.baseDir}assets/dummy.svg`,
          alt: " "
        })), title && _react.default.createElement("span", null, title)), isItems && _react.default.createElement(_list.MenuList, {
          items: items
        }), children), show && _react.default.createElement("div", {
          onClick: handleClick,
          className: "background-black"
        }));
      }
    }
  });

  /**********************
  INTERNAL MODULE: ./item
  **********************/

  ims.set('./item', {
    hash: 4263323519,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.MenuItem = void 0;
      var _react = require("react");
      var _icons = require("pragmate-ui/icons");
      const MenuItem = ({
        item
      }) => {
        const {
          title,
          link,
          icon
        } = item;
        return _react.default.createElement("li", {
          className: "pull-down-menu__item"
        }, _react.default.createElement(_icons.Icon, {
          icon: icon
        }), _react.default.createElement("a", {
          href: link
        }, title));
      };
      exports.MenuItem = MenuItem;
    }
  });

  /**********************
  INTERNAL MODULE: ./list
  **********************/

  ims.set('./list', {
    hash: 3770438484,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.MenuList = void 0;
      var _react = require("react");
      var _item = require("./item");
      const MenuList = ({
        items
      }) => {
        const output = items.map(item => _react.default.createElement(_item.MenuItem, {
          key: item.icon,
          item: item
        }));
        return _react.default.createElement("ul", {
          className: "pull-down-menu__list"
        }, output);
      };
      exports.MenuList = MenuList;
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./index",
    "from": "DropdownBottomMenu",
    "name": "DropdownBottomMenu"
  }];
  let DropdownBottomMenu = _exports.DropdownBottomMenu = void 0;

  // Module exports
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'DropdownBottomMenu') && (_exports.DropdownBottomMenu = DropdownBottomMenu = require ? require('./index').DropdownBottomMenu : value);
  };
  const __beyond_pkg = _exports.__beyond_pkg = __pkg;
  const hmr = _exports.hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  __pkg.initialise(ims);
});
