// pragmate-ui/progress-bar/progress-bar.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.ProgressBar = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "pragmate-ui@0.0.6/progress-bar"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2]]);
  brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/progress-bar');
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./index
  ***********************/

  ims.set('./index', {
    hash: 1193690543,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.ProgressBar = ProgressBar;
      var _react = require("react");
      /* bundle */
      function ProgressBar(props) {
        const {
          value,
          maxValue = 100,
          variant = 'primary'
        } = props;
        const percentage = value / maxValue * 100;
        const cls = variant ? `progress-bar__progress ${variant}` : 'progress-bar__progress';
        return _react.default.createElement("div", {
          className: "pui-progress-bar"
        }, _react.default.createElement("span", {
          className: cls,
          style: {
            width: `${percentage}%`
          }
        }, value, "%"));
      }
    }
  });

  /***********************
  INTERNAL MODULE: ./types
  ***********************/

  ims.set('./types', {
    hash: 2062575629,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./index",
    "from": "ProgressBar",
    "name": "ProgressBar"
  }];
  let ProgressBar = _exports.ProgressBar = void 0;

  // Module exports
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'ProgressBar') && (_exports.ProgressBar = ProgressBar = require ? require('./index').ProgressBar : value);
  };
  const __beyond_pkg = _exports.__beyond_pkg = __pkg;
  const hmr = _exports.hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  __pkg.initialise(ims);
});
