// pragmate-ui/empty/empty.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react", "pragmate-ui/icons"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.Empty = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "pragmate-ui@0.0.6/empty"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2], ['pragmate-ui/icons', dependency_3]]);
  brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/empty');
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./empty
  ***********************/

  ims.set('./empty', {
    hash: 2492106652,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Empty = Empty;
      var _react = require("react");
      var _icons = require("pragmate-ui/icons");
      /*bundle*/
      function Empty(props) {
        const cls = `${props.className ?? ''} pui-empty-section`;
        return _react.default.createElement("div", {
          className: cls
        }, _react.default.createElement("div", {
          className: "content"
        }, props.icon && _react.default.createElement(_icons.Icon, {
          icon: props.icon
        }), props.text && _react.default.createElement("h3", null, props.text), props.children && props.children));
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./empty",
    "from": "Empty",
    "name": "Empty"
  }];
  let Empty = _exports.Empty = void 0;

  // Module exports
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Empty') && (_exports.Empty = Empty = require ? require('./empty').Empty : value);
  };
  const __beyond_pkg = _exports.__beyond_pkg = __pkg;
  const hmr = _exports.hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  __pkg.initialise(ims);
});
