// pragmate-ui/empty/empty.browser.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/styles';
import * as dependency_2 from 'react';
import * as dependency_3 from 'pragmate-ui/icons';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"pragmate-ui@0.0.6/empty"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1],['react', dependency_2],['pragmate-ui/icons', dependency_3]]);

brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/empty')


const ims = new Map();

/***********************
INTERNAL MODULE: ./empty
***********************/

ims.set('./empty', {hash: 2492106652, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Empty = Empty;
var _react = require("react");
var _icons = require("pragmate-ui/icons");
/*bundle*/function Empty(props) {
  const cls = `${props.className ?? ''} pui-empty-section`;
  return _react.default.createElement("div", {
    className: cls
  }, _react.default.createElement("div", {
    className: "content"
  }, props.icon && _react.default.createElement(_icons.Icon, {
    icon: props.icon
  }), props.text && _react.default.createElement("h3", null, props.text), props.children && props.children));
}
}});

__pkg.exports.descriptor = [{"im":"./empty","from":"Empty","name":"Empty"}];

export let Empty;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Empty') && (Empty = require ? require('./empty').Empty : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
