// pragmate-ui/dropdown/dropdown.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.DropdownToggle = _exports.DropdownProvider = _exports.DropdownLink = _exports.DropDownMenu = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "pragmate-ui@0.0.6/dropdown"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2]]);
  brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/dropdown');
  const ims = new Map();

  /*************************
  INTERNAL MODULE: ./context
  *************************/

  ims.set('./context', {
    hash: 4270738753,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.DropdownProvider = DropdownProvider;
      exports.useDropdownContext = void 0;
      var _react = require("react");
      const DropdownContext = _react.default.createContext(null);
      const useDropdownContext = () => (0, _react.useContext)(DropdownContext);
      exports.useDropdownContext = useDropdownContext;
      /*bundle*/
      function DropdownProvider(props) {
        const {
          children,
          isToggle,
          onToggle,
          isClose
        } = props;
        const [showMenu, setShowMenu] = _react.default.useState(false);
        const value = {
          showMenu,
          setShowMenu,
          isToggle,
          onToggle,
          isClose
        };
        return _react.default.createElement(DropdownContext.Provider, {
          value: value
        }, children);
      }
    }
  });

  /**********************
  INTERNAL MODULE: ./link
  **********************/

  ims.set('./link', {
    hash: 3487775132,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.DropdownLink = DropdownLink;
      var _react = require("react");
      var _context = require("./context");
      /*bundle*/
      function DropdownLink(props) {
        const {
          setShowMenu
        } = (0, _context.useDropdownContext)();
        const {
          children,
          className,
          onClick,
          tag,
          id
        } = props;
        const handleClick = async event => {
          event.stopPropagation();
          if (!!onClick && typeof onClick === "function") await onClick(event);
          setShowMenu(false);
        };
        const TagLink = tag ?? "div";
        const cls = `${className ?? ""} dropdown-link`;
        const properties = Object.assign({}, props);
        delete properties.className;
        delete properties.onClick;
        delete properties.tag;
        delete properties.id;
        return _react.default.createElement(TagLink, {
          id: id,
          className: cls,
          onClick: handleClick,
          ...properties
        }, children);
      }
    }
  });

  /**********************
  INTERNAL MODULE: ./menu
  **********************/

  ims.set('./menu', {
    hash: 281379387,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.DropDownMenu = DropDownMenu;
      var _react = require("react");
      var _context = require("./context");
      /*bundle*/
      function DropDownMenu({
        children,
        className
      }) {
        const {
          showMenu,
          setShowMenu,
          isToggle,
          onToggle,
          isClose
        } = (0, _context.useDropdownContext)();
        const ref = _react.default.useRef(null);
        _react.default.useEffect(() => {
          const handleClick = event => {
            const {
              current
            } = ref;
            const isSameNode = current === event.target || current === event.currentTarget || event.composedPath()[0] === current;
            const isAChildren = current?.contains(event.composedPath()[0]);
            if (!isSameNode && !isAChildren) {
              setShowMenu(false);
              if (!!onToggle && isToggle && typeof onToggle === 'function' && isClose) {
                onToggle();
              }
            }
          };
          document.addEventListener('click', handleClick);
          return () => document.removeEventListener('click', handleClick);
        }, [isClose]);
        if (!showMenu && !isToggle) return null;
        const cls = `${className ?? ''} dropdown-menu-container`;
        return _react.default.createElement("div", {
          ref: ref,
          className: cls
        }, children);
      }
    }
  });

  /************************
  INTERNAL MODULE: ./toggle
  ************************/

  ims.set('./toggle', {
    hash: 2970189459,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.DropdownToggle = DropdownToggle;
      var React = require("react");
      var _context = require("./context");
      /*bundle*/
      function DropdownToggle({
        children,
        className
      }) {
        const cls = `${className ?? ''} dropdown-toggle`;
        const {
          showMenu,
          setShowMenu,
          isToggle,
          onToggle
        } = (0, _context.useDropdownContext)();
        const handleClick = event => {
          event.stopPropagation();
          if (!isToggle) setShowMenu(!showMenu);
          if (!!onToggle && isToggle && typeof onToggle === 'function') onToggle();
        };
        return React.createElement("div", {
          onClick: handleClick,
          className: cls
        }, children);
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./context",
    "from": "DropdownProvider",
    "name": "DropdownProvider"
  }, {
    "im": "./link",
    "from": "DropdownLink",
    "name": "DropdownLink"
  }, {
    "im": "./menu",
    "from": "DropDownMenu",
    "name": "DropDownMenu"
  }, {
    "im": "./toggle",
    "from": "DropdownToggle",
    "name": "DropdownToggle"
  }];
  let DropdownProvider = _exports.DropdownProvider = void 0,
    DropdownLink = _exports.DropdownLink = void 0,
    DropDownMenu = _exports.DropDownMenu = void 0,
    DropdownToggle = _exports.DropdownToggle = void 0;

  // Module exports
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'DropdownProvider') && (_exports.DropdownProvider = DropdownProvider = require ? require('./context').DropdownProvider : value);
    (require || prop === 'DropdownLink') && (_exports.DropdownLink = DropdownLink = require ? require('./link').DropdownLink : value);
    (require || prop === 'DropDownMenu') && (_exports.DropDownMenu = DropDownMenu = require ? require('./menu').DropDownMenu : value);
    (require || prop === 'DropdownToggle') && (_exports.DropdownToggle = DropdownToggle = require ? require('./toggle').DropdownToggle : value);
  };
  const __beyond_pkg = _exports.__beyond_pkg = __pkg;
  const hmr = _exports.hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  __pkg.initialise(ims);
});
