// pragmate-ui/chips/chips.browser.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/styles';
import * as dependency_2 from 'react';
import * as dependency_3 from 'pragmate-ui/ripple';
import * as dependency_4 from 'tippy.js';
import * as dependency_5 from 'pragmate-ui/icons';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"pragmate-ui@0.0.6/chips"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1],['react', dependency_2],['pragmate-ui/ripple', dependency_3],['tippy.js', dependency_4],['pragmate-ui/icons', dependency_5]]);

brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/chips')


const ims = new Map();

/***********************
INTERNAL MODULE: ./badge
***********************/

ims.set('./badge', {hash: 3444524587, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Chip = Chip;
var _react = require("react");
var _ripple = require("pragmate-ui/ripple");
var _tippy = require("tippy.js");
var _icons = require("pragmate-ui/icons");
/*bundle*/function Chip(props) {
  const {
    item,
    type,
    className,
    title,
    children,
    icon
  } = props;
  const properties = {
    ...props
  };
  const ref = _react.default.useRef(null);
  ['children', 'type', 'className'].forEach(key => delete properties[key]);
  _react.default.useEffect(() => {
    if (properties.onClick) {
      const ripple = new _ripple.RippleEffect();
      ripple.addRippleEffect(ref.current);
    }
    if (title) (0, _tippy.default)(ref.current);
  }, [properties.onClick, title]);
  if (title) properties['data-tippy-content'] = title;
  let cls = `pui-chip${type ? ` pui-chip--${type}` : ''}${className ? ` ${className}` : ''}`;
  if (properties.onClick) {
    cls += ` is-clickable`;
  }
  const showIcon = icon && _react.default.createElement(_icons.Icon, {
    icon: icon
  });
  return _react.default.createElement("span", {
    ref: ref,
    className: cls,
    ...properties
  }, item, children, showIcon);
}
}});

/***********************
INTERNAL MODULE: ./index
***********************/

ims.set('./index', {hash: 1542096823, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Chips = Chips;
var _react = require("react");
/*bundle*/function Chips(props) {
  const {
    children
  } = props;
  const properties = {
    ...props
  };
  ['items', 'type'].forEach(key => delete properties[key]);
  const cls = `${props.type ? `pui-chip__container--${props.type}` : ''}`;
  properties.className = `pui-chip__container${properties.className ? ` ${properties.className}` : ''} ${cls}`;
  return _react.default.createElement("div", {
    ...properties
  }, _react.default.createElement("div", {
    className: 'content-chip'
  }, children));
}
}});

__pkg.exports.descriptor = [{"im":"./badge","from":"Chip","name":"Chip"},{"im":"./index","from":"Chips","name":"Chips"}];

export let Chip, Chips;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Chip') && (Chip = require ? require('./badge').Chip : value);
    (require || prop === 'Chips') && (Chips = require ? require('./index').Chips : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
