// pragmate-ui/chips/chips.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react", "pragmate-ui/ripple", "tippy.js", "pragmate-ui/icons"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2, dependency_3, dependency_4, dependency_5) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.Chips = _exports.Chip = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "pragmate-ui@0.0.6/chips"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2], ['pragmate-ui/ripple', dependency_3], ['tippy.js', dependency_4], ['pragmate-ui/icons', dependency_5]]);
  brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/chips');
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./badge
  ***********************/

  ims.set('./badge', {
    hash: 3444524587,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Chip = Chip;
      var _react = require("react");
      var _ripple = require("pragmate-ui/ripple");
      var _tippy = require("tippy.js");
      var _icons = require("pragmate-ui/icons");
      /*bundle*/
      function Chip(props) {
        const {
          item,
          type,
          className,
          title,
          children,
          icon
        } = props;
        const properties = {
          ...props
        };
        const ref = _react.default.useRef(null);
        ['children', 'type', 'className'].forEach(key => delete properties[key]);
        _react.default.useEffect(() => {
          if (properties.onClick) {
            const ripple = new _ripple.RippleEffect();
            ripple.addRippleEffect(ref.current);
          }
          if (title) (0, _tippy.default)(ref.current);
        }, [properties.onClick, title]);
        if (title) properties['data-tippy-content'] = title;
        let cls = `pui-chip${type ? ` pui-chip--${type}` : ''}${className ? ` ${className}` : ''}`;
        if (properties.onClick) {
          cls += ` is-clickable`;
        }
        const showIcon = icon && _react.default.createElement(_icons.Icon, {
          icon: icon
        });
        return _react.default.createElement("span", {
          ref: ref,
          className: cls,
          ...properties
        }, item, children, showIcon);
      }
    }
  });

  /***********************
  INTERNAL MODULE: ./index
  ***********************/

  ims.set('./index', {
    hash: 1542096823,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Chips = Chips;
      var _react = require("react");
      /*bundle*/
      function Chips(props) {
        const {
          children
        } = props;
        const properties = {
          ...props
        };
        ['items', 'type'].forEach(key => delete properties[key]);
        const cls = `${props.type ? `pui-chip__container--${props.type}` : ''}`;
        properties.className = `pui-chip__container${properties.className ? ` ${properties.className}` : ''} ${cls}`;
        return _react.default.createElement("div", {
          ...properties
        }, _react.default.createElement("div", {
          className: 'content-chip'
        }, children));
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./badge",
    "from": "Chip",
    "name": "Chip"
  }, {
    "im": "./index",
    "from": "Chips",
    "name": "Chips"
  }];
  let Chip = _exports.Chip = void 0,
    Chips = _exports.Chips = void 0;

  // Module exports
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Chip') && (_exports.Chip = Chip = require ? require('./badge').Chip : value);
    (require || prop === 'Chips') && (_exports.Chips = Chips = require ? require('./index').Chips : value);
  };
  const __beyond_pkg = _exports.__beyond_pkg = __pkg;
  const hmr = _exports.hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  __pkg.initialise(ims);
});
