// pragmate-ui/badge/badge.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/styles';
import * as dependency_2 from 'react';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"pragmate-ui@0.0.6/badge"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1],['react', dependency_2]]);

brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/badge')


const ims = new Map();

/***********************
INTERNAL MODULE: ./index
***********************/

ims.set('./index', {hash: 2882574013, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Badge = Badge;
var _react = require("react");
/*bundle*/function Badge({
  children,
  value,
  visible = false,
  variant = 'default',
  dot
}) {
  const classes = ['primary', 'secondary', 'success', 'danger', 'warning', 'info', 'default'];
  let cls = `pui-badge${classes.includes(variant) ? ` pui-badge--${variant}` : ''}`;
  if (visible) cls += ' pui-badge--visible';
  if (value) cls += ' pui-badge--has-value';
  if (dot) cls += ' dot';
  return _react.default.createElement("div", {
    className: 'pui-badge__container'
  }, _react.default.createElement("div", {
    className: 'content-children'
  }, children), _react.default.createElement("span", {
    className: cls
  }, value));
}
}});

__pkg.exports.descriptor = [{"im":"./index","from":"Badge","name":"Badge"}];

export let Badge;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Badge') && (Badge = require ? require('./index').Badge : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
