// pragmate-ui/tooltip/tooltip.cjs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.hmr = exports.__beyond_pkg = exports.Tooltip = void 0;
var dependency_0 = require("@beyond-js/kernel/bundle");
var dependency_1 = require("react");
var dependency_2 = require("tippy.js");
const {
  Bundle: __Bundle
} = dependency_0;
const __pkg = new __Bundle({
  "module": {
    "vspecifier": "pragmate-ui@0.0.6/tooltip"
  },
  "type": "ts"
}).package();
;
__pkg.dependencies.update([['react', dependency_1], ['tippy.js', dependency_2]]);
const ims = new Map();

/********************************
INTERNAL MODULE: ./ITippySettings
********************************/

ims.set('./ITippySettings', {
  hash: 3730348027,
  creator: function (require, exports) {
    "use strict";

    Object.defineProperty(exports, "__esModule", {
      value: true
    });
  }
});

/***********************
INTERNAL MODULE: ./index
***********************/

ims.set('./index', {
  hash: 780162422,
  creator: function (require, exports) {
    "use strict";

    Object.defineProperty(exports, "__esModule", {
      value: true
    });
    exports.Tooltip = Tooltip;
    var _react = require("react");
    var _tippy = require("tippy.js");
    /*bundle*/
    function Tooltip({
      children,
      content,
      placement = 'top',
      settings = {}
    }) {
      const ref = _react.default.useRef(null);
      const placements = ['top', 'top-start', 'top-end', 'right', 'right-start', 'right-end', 'bottom', 'bottom-start', 'bottom-end', 'left', 'left-start', 'left-end', 'auto', 'auto-start', 'auto-end'];
      if (!placements.includes(placement)) placement = 'top';
      const specs = {
        placement,
        content
      };
      _react.default.useEffect(() => {
        (0, _tippy.default)(ref.current, specs);
      }, []);
      return _react.default.createElement("span", {
        ref: ref,
        className: "pui-tooltip"
      }, children);
    }
  }
});
__pkg.exports.descriptor = [{
  "im": "./index",
  "from": "Tooltip",
  "name": "Tooltip"
}];
let Tooltip = exports.Tooltip = void 0;

// Module exports
__pkg.exports.process = function ({
  require,
  prop,
  value
}) {
  (require || prop === 'Tooltip') && (exports.Tooltip = Tooltip = require ? require('./index').Tooltip : value);
};
const __beyond_pkg = exports.__beyond_pkg = __pkg;
const hmr = exports.hmr = new function () {
  this.on = (event, listener) => void 0;
  this.off = (event, listener) => void 0;
}();
__pkg.initialise(ims);
