// pragmate-ui/progress-bar/progress-bar.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/styles';
import * as dependency_2 from 'react';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"pragmate-ui@0.0.6/progress-bar"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1],['react', dependency_2]]);

brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/progress-bar')


const ims = new Map();

/***********************
INTERNAL MODULE: ./index
***********************/

ims.set('./index', {hash: 1193690543, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ProgressBar = ProgressBar;
var _react = require("react");
/* bundle */function ProgressBar(props) {
  const {
    value,
    maxValue = 100,
    variant = 'primary'
  } = props;
  const percentage = value / maxValue * 100;
  const cls = variant ? `progress-bar__progress ${variant}` : 'progress-bar__progress';
  return _react.default.createElement("div", {
    className: "pui-progress-bar"
  }, _react.default.createElement("span", {
    className: cls,
    style: {
      width: `${percentage}%`
    }
  }, value, "%"));
}
}});

/***********************
INTERNAL MODULE: ./types
***********************/

ims.set('./types', {hash: 2062575629, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
}});

__pkg.exports.descriptor = [{"im":"./index","from":"ProgressBar","name":"ProgressBar"}];

export let ProgressBar;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'ProgressBar') && (ProgressBar = require ? require('./index').ProgressBar : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
