// pragmate-ui/tabs/tabs.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "@beyond-js/kernel/styles", "react"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.TabsContainer = _exports.Tabs = _exports.Tab = _exports.Panes = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "pragmate-ui@0.0.6/tabs"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1], ['react', dependency_2]]);
  brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/tabs');
  const ims = new Map();

  /*************************
  INTERNAL MODULE: ./context
  *************************/

  ims.set('./context', {
    hash: 1113529643,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.useTabsContext = exports.TabsContext = void 0;
      var _react = require("react");
      const TabsContext = exports.TabsContext = _react.default.createContext({});
      const useTabsContext = () => _react.default.useContext(TabsContext);
      exports.useTabsContext = useTabsContext;
    }
  });

  /***********************
  INTERNAL MODULE: ./index
  ***********************/

  ims.set('./index', {
    hash: 1605311872,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.TabsContainer = void 0;
      var _react = require("react");
      var _context = require("./context");
      /* bundle  */
      const TabsContainer = ({
        active,
        onChange,
        className,
        children,
        panes
      }) => {
        const activeIndex = active || 0;
        const [activeTab, setActiveTab] = _react.default.useState(activeIndex);
        const value = {
          panes,
          activeTab,
          setActiveTab,
          onChange
        };
        const cls = `pui-tabs-container${className ? ` ${className}` : ''}`;
        return _react.default.createElement(_context.TabsContext.Provider, {
          value: value
        }, _react.default.createElement("section", {
          className: cls
        }, children));
      };
      exports.TabsContainer = TabsContainer;
    }
  });

  /****************************
  INTERNAL MODULE: ./interfaces
  ****************************/

  ims.set('./interfaces', {
    hash: 1404915735,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
    }
  });

  /***********************
  INTERNAL MODULE: ./panes
  ***********************/

  ims.set('./panes', {
    hash: 2151215637,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Panes = Panes;
      var _react = require("react");
      var _context = require("./context");
      /*bundle*/
      function Panes({
        children,
        className
      }) {
        const {
          activeTab
        } = (0, _context.useTabsContext)();
        const childrenArray = _react.default.Children.toArray(children);
        const Control = childrenArray[activeTab];
        const cls = `tab-content ${className ? ` ${className}` : ''}`;
        return _react.default.createElement("section", {
          className: cls
        }, Control);
      }
    }
  });

  /*********************
  INTERNAL MODULE: ./tab
  *********************/

  ims.set('./tab', {
    hash: 4224080064,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Tab = Tab;
      var _react = require("react");
      var _context = require("./context");
      /*bundle */
      function Tab(props) {
        const {
          children,
          index,
          disabled
        } = props;
        const {
          activeTab,
          setActiveTab,
          onChange
        } = (0, _context.useTabsContext)();
        const cls = `tab ${index === activeTab ? 'active' : ''} ${disabled ? ' is-disabled' : ''}`;
        const onClick = event => {
          if (onChange) onChange(event, index);
          setActiveTab(index);
        };
        const attrs = {};
        if (!disabled) attrs.onClick = onClick;
        return _react.default.createElement("div", {
          className: cls,
          ...attrs
        }, children);
      }
    }
  });

  /**********************
  INTERNAL MODULE: ./tabs
  **********************/

  ims.set('./tabs', {
    hash: 322980779,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Tabs = void 0;
      var _react = require("react");
      var _context = require("./context");
      var _useScroll = require("./use-scroll");
      var _tab = require("./tab");
      /*bundle*/
      const Tabs = ({
        children,
        className
      }) => {
        const {
          activeTab
        } = (0, _context.useTabsContext)();
        const ref = _react.default.useRef(null);
        (0, _useScroll.useScroll)(ref, activeTab);
        const output = children.map((item, index) => {
          const props = {
            ...item.props,
            index,
            key: index
          };
          if (_react.default.isValidElement(item) && item.type === _tab.Tab) {
            const tabChild = item;
            return _react.default.cloneElement(tabChild, props);
          }
          return item;
        });
        const cls = `pui-tabs-menu${className ? ` ${className}` : ''}`;
        return _react.default.createElement("header", {
          className: cls
        }, _react.default.createElement("div", {
          className: 'tabs',
          ref: ref
        }, output));
      };
      exports.Tabs = Tabs;
    }
  });

  /****************************
  INTERNAL MODULE: ./use-scroll
  ****************************/

  ims.set('./use-scroll', {
    hash: 1668718188,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.useScroll = useScroll;
      var _react = require("react");
      function useScroll(tabRef, activeTab) {
        _react.default.useEffect(() => {
          const scrollToTab = index => {
            const tabElement = tabRef.current?.querySelectorAll('.tab')[index];
            if (tabElement) {
              const container = tabRef.current;
              const containerRect = container.getBoundingClientRect();
              const tabRect = tabElement.getBoundingClientRect();
              const scrollLeft = container.scrollLeft;
              if (tabRect.left < containerRect.left) {
                container.scrollTo({
                  left: scrollLeft + tabRect.left - containerRect.left,
                  behavior: 'smooth'
                });
              } else if (tabRect.right > containerRect.right) {
                container.scrollTo({
                  left: scrollLeft + tabRect.right - containerRect.right,
                  behavior: 'smooth'
                });
              }
            }
          };
          scrollToTab(activeTab);
        }, [activeTab]);
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./index",
    "from": "TabsContainer",
    "name": "TabsContainer"
  }, {
    "im": "./panes",
    "from": "Panes",
    "name": "Panes"
  }, {
    "im": "./tab",
    "from": "Tab",
    "name": "Tab"
  }, {
    "im": "./tabs",
    "from": "Tabs",
    "name": "Tabs"
  }];
  let TabsContainer = _exports.TabsContainer = void 0,
    Panes = _exports.Panes = void 0,
    Tab = _exports.Tab = void 0,
    Tabs = _exports.Tabs = void 0;

  // Module exports
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'TabsContainer') && (_exports.TabsContainer = TabsContainer = require ? require('./index').TabsContainer : value);
    (require || prop === 'Panes') && (_exports.Panes = Panes = require ? require('./panes').Panes : value);
    (require || prop === 'Tab') && (_exports.Tab = Tab = require ? require('./tab').Tab : value);
    (require || prop === 'Tabs') && (_exports.Tabs = Tabs = require ? require('./tabs').Tabs : value);
  };
  const __beyond_pkg = _exports.__beyond_pkg = __pkg;
  const hmr = _exports.hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  __pkg.initialise(ims);
});
