// pragmate-ui/link/link.amd.js
define(["exports", "module", "@beyond-js/kernel/bundle", "react", "@beyond-js/kernel/routing"], function (_exports, _amd_module, dependency_0, dependency_1, dependency_2) {
  "use strict";

  Object.defineProperty(_exports, "__esModule", {
    value: true
  });
  _exports.hmr = _exports.__beyond_pkg = _exports.Link = void 0;
  const {
    Bundle: __Bundle
  } = dependency_0;
  const __pkg = new __Bundle({
    "module": {
      "vspecifier": "pragmate-ui@0.0.6/link"
    },
    "type": "code"
  }, _amd_module.uri).package();
  ;
  __pkg.dependencies.update([['react', dependency_1], ['@beyond-js/kernel/routing', dependency_2]]);
  const ims = new Map();

  /***********************
  INTERNAL MODULE: ./index
  ***********************/

  ims.set('./index', {
    hash: 3979789248,
    creator: function (require, exports) {
      "use strict";

      Object.defineProperty(exports, "__esModule", {
        value: true
      });
      exports.Link = Link;
      var _react = require("react");
      var _routing = require("@beyond-js/kernel/routing");
      /*bundle*/
      function Link({
        href,
        ...props
      }) {
        const isExternal = props.target === '_blank';
        const onClick = event => {
          if (!isExternal) event.preventDefault();
          event.stopPropagation();
          if (props.onClick && typeof props.onClick === 'function') {
            props.onClick(event);
          }
          !isExternal && _routing.routing.pushState(href);
        };
        const properties = {
          ...props
        };
        ['href', 'onClick'].forEach(prop => delete properties[prop]);
        return _react.default.createElement("a", {
          ...properties,
          href: href,
          onClick: onClick,
          target: props.target
        }, props.children);
      }
    }
  });
  __pkg.exports.descriptor = [{
    "im": "./index",
    "from": "Link",
    "name": "Link"
  }];
  let Link = _exports.Link = void 0;

  // Module exports
  __pkg.exports.process = function ({
    require,
    prop,
    value
  }) {
    (require || prop === 'Link') && (_exports.Link = Link = require ? require('./index').Link : value);
  };
  const __beyond_pkg = _exports.__beyond_pkg = __pkg;
  const hmr = _exports.hmr = new function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
  }();
  __pkg.initialise(ims);
});
