// pragmate-ui/link/link.browser.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from 'react';
import * as dependency_2 from '@beyond-js/kernel/routing';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"pragmate-ui@0.0.6/link"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['react', dependency_1],['@beyond-js/kernel/routing', dependency_2]]);

const ims = new Map();

/***********************
INTERNAL MODULE: ./index
***********************/

ims.set('./index', {hash: 3979789248, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Link = Link;
var _react = require("react");
var _routing = require("@beyond-js/kernel/routing");
/*bundle*/function Link({
  href,
  ...props
}) {
  const isExternal = props.target === '_blank';
  const onClick = event => {
    if (!isExternal) event.preventDefault();
    event.stopPropagation();
    if (props.onClick && typeof props.onClick === 'function') {
      props.onClick(event);
    }
    !isExternal && _routing.routing.pushState(href);
  };
  const properties = {
    ...props
  };
  ['href', 'onClick'].forEach(prop => delete properties[prop]);
  return _react.default.createElement("a", {
    ...properties,
    href: href,
    onClick: onClick,
    target: props.target
  }, props.children);
}
}});

__pkg.exports.descriptor = [{"im":"./index","from":"Link","name":"Link"}];

export let Link;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Link') && (Link = require ? require('./index').Link : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
