// pragmate-ui/toast/toast.browser.mjs
import * as dependency_0 from '@beyond-js/kernel/bundle';
import * as dependency_1 from '@beyond-js/kernel/styles';
import * as dependency_2 from 'react';
import * as dependency_3 from '@beyond-js/react-18-widgets/hooks';
import * as dependency_4 from '@beyond-js/reactive/model';
import * as dependency_5 from 'pragmate-ui/icons';

const {Bundle: __Bundle} = dependency_0;
const __pkg = new __Bundle({"module":{"vspecifier":"pragmate-ui@0.0.6/toast"},"type":"code"}, import.meta.url).package();;

__pkg.dependencies.update([['@beyond-js/kernel/styles', dependency_1],['react', dependency_2],['@beyond-js/react-18-widgets/hooks', dependency_3],['@beyond-js/reactive/model', dependency_4],['pragmate-ui/icons', dependency_5]]);

brequire('@beyond-js/kernel/styles').styles.register('pragmate-ui@0.0.6/toast')


const ims = new Map();

/***********************
INTERNAL MODULE: ./index
***********************/

ims.set('./index', {hash: 182626329, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Toasts = Toasts;
var _react = require("react");
var _hooks = require("@beyond-js/react-18-widgets/hooks");
var _model = require("./model");
var _toast = require("./toast");
/*bundle*/function Toasts({
  position = {
    bottom: '1rem',
    right: '1rem'
  },
  className,
  ...props
}) {
  const [items, setItems] = _react.default.useState([]);
  (0, _hooks.useBinder)([_model.toast], () => setItems(_model.toast.current), 'current.toast.changed');
  const elements = items.map(item => _react.default.createElement(_toast.Toast, {
    key: item.id,
    ...item
  }));
  return _react.default.createElement("div", {
    style: position,
    className: `pui-toast__container`,
    ...props
  }, elements);
}
}});

/***********************
INTERNAL MODULE: ./model
***********************/

ims.set('./model', {hash: 1270696975, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.toast = void 0;
var _model = require("@beyond-js/reactive/model");
class Toast extends _model.ReactiveModel {
  #current;
  get current() {
    return this.#current;
  }
  set current(newValue) {
    this.#current = newValue;
    this.triggerEvent('current.toast.changed');
  }
  constructor() {
    super();
    this.#current = [];
  }
  #generateUniqueId() {
    return crypto.getRandomValues(new Uint8Array(16)).toString();
  }
  #add(type, message, duration) {
    const newToast = {
      id: this.#generateUniqueId(),
      message,
      type,
      duration
    };
    this.#current = [...this.#current, newToast];
    this.triggerEvent('current.toast.changed');
    return newToast.id;
  }
  remove(toastId) {
    this.#current = this.#current.filter(toast => toast.id !== toastId);
    this.triggerEvent('current.toast.changed');
  }
  success(message, duration) {
    return this.#add('success', message, duration);
  }
  error(message, duration) {
    return this.#add('error', message, duration);
  }
  warning(message, duration) {
    return this.#add('warning', message, duration);
  }
  info(message, duration) {
    return this.#add('info', message, duration);
  }
  loading(message, duration) {
    return this.#add('loading', message, duration);
  }
}
/*bundle*/const toast = exports.toast = new Toast();
}});

/***********************
INTERNAL MODULE: ./toast
***********************/

ims.set('./toast', {hash: 2244319811, creator: function (require, exports) {
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Toast = Toast;
var _react = require("react");
var _model = require("./model");
var _icons = require("pragmate-ui/icons");
const DEFAULT_DURATION = 3000;
const ANIMATION_MARGIN = 300;
/*bundle*/function Toast({
  type,
  message,
  duration,
  id
}) {
  const toastRef = _react.default.useRef(null);
  _react.default.useEffect(() => {
    if (toastRef.current) {
      toastRef.current.classList.add('enter');
    }
    return () => {
      if (toastRef.current) toastRef.current.classList.remove('enter');
    };
  }, [toastRef.current]);
  _react.default.useEffect(() => {
    duration = duration ?? DEFAULT_DURATION;
    setTimeout(() => {
      toastRef.current.classList.remove('enter');
      toastRef.current.classList.add('exit');
    }, duration);
    const timeout = setTimeout(() => {
      _model.toast.current = _model.toast.current.filter(item => item.id !== id);
    }, duration + ANIMATION_MARGIN);
    return () => clearTimeout(timeout);
  }, [duration, id]);
  const icons = {
    error: 'triangle-exclamation',
    success: 'tickCircle',
    info: 'infoCircle',
    loading: 'refreshCircle',
    warning: 'warning'
  };
  return _react.default.createElement("article", {
    ref: toastRef,
    className: `toast ${type}`
  }, _react.default.createElement(_icons.Icon, {
    icon: icons[type],
    className: 'icon'
  }), _react.default.createElement("p", {
    className: 'message'
  }, message));
}
}});

__pkg.exports.descriptor = [{"im":"./index","from":"Toasts","name":"Toasts"},{"im":"./model","from":"ToastTypes","name":"ToastTypes"},{"im":"./model","from":"IToast","name":"IToast"},{"im":"./model","from":"toast","name":"toast"},{"im":"./toast","from":"Toast","name":"Toast"}];

export let Toasts, ToastTypes, IToast, toast, Toast;

// Module exports
__pkg.exports.process = function({require, prop, value}) {
    (require || prop === 'Toasts') && (Toasts = require ? require('./index').Toasts : value);
    (require || prop === 'ToastTypes') && (ToastTypes = require ? require('./model').ToastTypes : value);
    (require || prop === 'IToast') && (IToast = require ? require('./model').IToast : value);
    (require || prop === 'toast') && (toast = require ? require('./model').toast : value);
    (require || prop === 'Toast') && (Toast = require ? require('./toast').Toast : value);

};
export const __beyond_pkg = __pkg;

export const hmr = new (function () {
    this.on = (event, listener) => void 0;
    this.off = (event, listener) => void 0;
});


__pkg.initialise(ims);
